app/services/external/gpt_service.py

```
@@ -38,6 +39,50 @@ def infer_brand_name_from_keywords(keywords: list[str]) -> str:
+            # 코드블럭 제거
+            if message.startswith("```json"):
+                message = message[7:]
+            if message.startswith("```"):
+                message = message[3:]
+            if message.endswith("```"):
+                message = message[:-3]
+            return json.loads(message)
+
+        except json.JSONDecodeError as e:
+            logger.error(f"[GPTService] JSON 파싱 실패: {e}")
+            raise APIException(ErrorStatus.GPT_RESPONSE_PARSE_ERROR)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)